compiler/rustc_resolve/src/imports.rs

```diff
@@ -931,6 +935,23 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                 _ => unreachable!(),
             };
 
+        let mut res = None;
+        if !self.privacy_errors.is_empty() {
+            // Get the Res for the last element, so that we can point to alternative ways of
+            // importing it if available.
+            let mut path = import.module_path.clone();
+            path.push(Segment::from_ident(ident));
+            if let PathResult::Module(ModuleOrUniformRoot::Module(module)) =
+                self.resolve_path(&path, None, &import.parent_scope, None, ignore_binding)
```

Comment:
```suggestion
                self.resolve_path(&path, None, &import.parent_scope, Some(finalize), ignore_binding)
```
Btw, what happens if you use the same `finalize` as above (with `ReportPrivate::Report`).
The dedup span will be the same as in resolution above (`import.span`), so the reported errors should be deduplicated into nothing?

Reply:
This actually solves the issue I was seeing in https://github.com/rust-lang/rust/pull/112686#discussion_r1237360321. Changed.